### PR TITLE
feat: Add address balance format toggle

### PR DESCRIPTION
**Pull request by abbasabid**
# Description of change

- Address balance toggle.
- Add copy button to copy the balance.

Aims to complete https://github.com/iotaledger/explorer/issues/312

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## Change checklist

- [x] My code follows the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: client/src/app/components/chrysalis/TransactionPayload.tsx [hunk 1]
```diff
@@ -28,7 +30,8 @@ class TransactionPayload extends AsyncComponent<TransactionPayloadProps, Transac
 
         this.state = {
             showInputDetails: -1,
-            showOutputDetails: -1
+            showOutputDetails: -1,
+            toggleBalance: false
```
Comment (msarcev) [inline client/src/app/components/chrysalis/TransactionPayload.tsx:34]: Should we rename this to something like `isFormattedBalance` ?

**Review by msarcev — CHANGES_REQUESTED** [review]
Looks fine, i just found a couple of minor UI/UX issues:
I the message outputs part, when we use the copy button, the "Copied" that flashes looks like this:
image
On the address page, the copy button has different css then on message outputs:
- On click doesn't make it green and it has a green border instead. Also the copied message is black.
image

File: client/src/app/components/chrysalis/TransactionPayload.tsx (hunk 1 above)
Comment (abbasabid) [reply client/src/app/components/chrysalis/TransactionPayload.tsx]: Done.

**Comment by abbasabid** [comment]
> Looks fine, i just found a couple of minor UI/UX issues.
> 
> I the message outputs part, when we use the copy button, the "Copied" that flashes looks like this: image
> 
> On the address page, the copy button has different css then on message outputs. On click doesn't make it green and it has a green border instead. Also the copied message is black. image

Fixed.

File: client/src/app/components/MessageButton.scss [hunk 2]
```diff
@@ -46,6 +46,16 @@
     }
   }
 
+  .message-label--bottom {
```
Comment (msarcev) [inline client/src/app/components/MessageButton.scss:49]: What's the difference between `.message-button--message` and `.message-label` ? 
Looks to me like they are the same thing. So why call them differently ?
This should probably be just `bottom` and `top` and nested inside `.message-button--message` as it's always attached after that class, never alone.

File: client/src/app/routes/chrysalis/Message.tsx [hunk 3]
```diff
@@ -173,7 +173,7 @@ class Message extends AsyncComponent<RouteComponentProps<MessageRouteProps>, Mes
                                                 this.state.paramMessageId
                                             )}
                                             buttonType="copy"
-                                            labelPosition="top"
+                                            labelPosition="right"
```
Comment (msarcev) [inline client/src/app/routes/chrysalis/Message.tsx:176]: We should make it "right" by default if no prop is passed, so we can remove the prop from all these places.

File: client/src/app/components/MessageButton.scss (hunk 2 above)
Comment (abbasabid) [reply client/src/app/components/MessageButton.scss:49]: `.message-button--message` has the complete styles for the copy button like its color, animation and transform etc and `.message-label--bottom` and `.message-label--top` are only used to override the `.message-button--message` styles for `top` and `right` properties.
I can change there name but it cannot be nested imho.

File: client/src/app/routes/chrysalis/Message.tsx (hunk 3 above)
Comment (abbasabid) [reply client/src/app/routes/chrysalis/Message.tsx]: Done.

File: client/src/app/components/MessageButton.scss (hunk 2 above)
Comment (msarcev) [reply client/src/app/components/MessageButton.scss]: Yes, you can nest them with the `&` scss operator. I think nesting them is better cause it makes it clearer that those classes are tied together.

**Pull request merged**